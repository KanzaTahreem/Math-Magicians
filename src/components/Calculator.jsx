import React from 'react';
import styles from '../stylesheets/Calculator.module.css';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = (e) => {
    this.setState((state) => calculate(state, e.target.textContent));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <section className={styles.calculator}>
        <div className={styles.container}>
          <div className={styles.display}>
            {total}
            {operation}
            {next}
          </div>
          <div className={styles.keys}>
            <div>
              <button type="button" className={styles.btn} onClick={this.handleClick}>AC</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>+/-</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>%</button>
              <button type="button" className={`${styles.btn} ${styles['btn-right']}`} onClick={this.handleClick}>รท</button>
            </div>
            <div>
              <button type="button" className={styles.btn} onClick={this.handleClick}>7</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>8</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>9</button>
              <button type="button" className={`${styles.btn} ${styles['btn-right']}`} onClick={this.handleClick}>x</button>
            </div>
            <div>
              <button type="button" className={styles.btn} onClick={this.handleClick}>4</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>5</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>6</button>
              <button type="button" className={`${styles.btn} ${styles['btn-right']}`} onClick={this.handleClick}>-</button>
            </div>
            <div>
              <button type="button" className={styles.btn} onClick={this.handleClick}>1</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>2</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>3</button>
              <button type="button" className={`${styles.btn} ${styles['btn-right']}`} onClick={this.handleClick}>+</button>
            </div>
            <div>
              <button type="button" className={`${styles.btn} ${styles['btn-expand']}`} onClick={this.handleClick}>0</button>
              <button type="button" className={styles.btn} onClick={this.handleClick}>.</button>
              <button type="button" className={`${styles.btn} ${styles['btn-right']}`} onClick={this.handleClick}>=</button>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default Calculator;
