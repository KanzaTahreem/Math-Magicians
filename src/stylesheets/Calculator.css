.cal-container {
  height: calc(100vh - 15vh);
  display: flex;
  width: 85vw;
  margin: 0 auto;
  justify-content: space-between;
}

.cal-container h2 {
  align-self: center;
  letter-spacing: 5px;
  font-size: 3em;
  margin-top: -3em;
  color: lightgray;
  text-shadow: 2px 2px 5px #2f2e2e;
}

.calculator {
  align-self: center;
}

.calculator > div {
  width: 30em;
  padding: 1em;
  height: 70vh;
  display: flex;
  border-radius: 10px;
  flex-direction: column;
  background: linear-gradient(140deg, #cdd3d7db, #eee8e8);
  box-shadow: 4px 4px 10px #c1c1c1, -4px -4px 10px #fff;
}

.display {
  width: 100%;
  display: flex;
  height: 15vh;
  padding: 1em;
  font-size: 2em;
  font-weight: 700;
  color: #f94892;
  align-items: center;
  border-radius: 10px;
  margin-bottom: 1em;
  justify-content: flex-end;
  background-color: #eeeded;
  text-shadow: 1px 1px 2px #000;
  box-shadow: inset 4px 4px 4px #c1c1c1, -4px -4px 4px #ddd;
}

.display > p {
  border-radius: 10px;
}

.keys {
  width: 100%;
  height: 100%;
  gap: 1em;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
}

.btn {
  font-family: 'Orbitron', sans-serif;
  font-weight: 600;
  font-size: 1.1em;
  border: #c1c1c1;
  border-radius: 10px;
  background: #eeeeeea8;
  box-shadow: 4px 4px 10px #c1c1c1;
  width: calc(25% - 1em);
}

.btn:hover {
  cursor: pointer;
  background-color: #f94892;
  color: #fff;
  transition: all 0.5s ease-in;
}

.btn:nth-child(17) {
  width: calc(50% - 1em);
}

.btn:nth-child(4),
.btn:nth-child(8),
.btn:nth-child(12),
.btn:nth-child(16),
.btn:nth-child(19) {
  background-color: #f94892;
}

.btn:nth-child(4):hover,
.btn:nth-child(8):hover,
.btn:nth-child(12):hover,
.btn:nth-child(16):hover,
.btn:nth-child(19):hover {
  background-color: #eee;
  color: #f94892;
}
