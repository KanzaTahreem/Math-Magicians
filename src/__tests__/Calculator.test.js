import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const calculator = renderer
    .create(<Calculator />)
    .toJSON();
  expect(calculator).toMatchSnapshot();
});
