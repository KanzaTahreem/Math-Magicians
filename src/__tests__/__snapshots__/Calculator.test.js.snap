// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="cal-container"
>
  <h2>
    Lets do some
     
    <i>
       maths!
    </i>
  </h2>
  <div
    className="calculator"
  >
    <div
      className="container"
    >
      <div
        className="display"
      />
      <div
        className="keys"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</section>
`;
